### PR TITLE
Add docs for V2

### PR DESCRIPTION
**Pull request by pradeepnschrodinger**
## Description

This adds documentations for the v2 release. 
I added a separate section for Plugins and Public API.
Currently the docs for Public API methods/state are written manually, but in the future we want this to be auto-generated.

## Screenshots:
image
image
image
image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by AmanGupta2708** [comment]
How will user switch between v1 and v2 docs ?

**Comment by pradeepnschrodinger** [comment]
> How will user switch between v1 and v2 docs ?

They can't. We never had this feature implemented. :(
With that said, upgrading from v1 to v2 should be non-breaking for clients.

File: build_helpers/buildAPIDocs.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ var FILES_TO_READ = [
   },
   {
     path: path.join(PROJECT_ROOT, 'src/FixedDataTableCellDefault.js'),
-    name: 'Cell',
+    name: 'DataCell',
```
Comment (karry08) [inline build_helpers/buildAPIDocs.sh:34]: is the component renamed or it was mistyped here

File: site/docs/DocsPage.js [hunk 2]
```diff
@@ -16,29 +16,45 @@ const DOCS_MARKDOWN_FILES = {
...
+  [DocsPages.PUBLIC_API.API_REFERENCE
+    .location]: require('../../docs/public_api/api_reference.md'),
 };
 
 class DocsPage extends React.Component {
   render() {
     const HTML = DOCS_MARKDOWN_FILES[this.props.page.location];
 
+    let pageClassname = 'pageBody';
+    if (this.props.page.pageClassname) {
```
Comment (karry08) [inline site/docs/DocsPage.js:49]: is page always defined here

**Comment by karry08** [comment]
How does the auto-generate thing works?

**Comment by karry08** [comment]
LGTM

File: build_helpers/buildAPIDocs.sh (hunk 1 above)
Comment (pradeepnschrodinger) [reply build_helpers/buildAPIDocs.sh:34]: `Cell` is the older and now deprecated component.
`DataCell` is the newer replacement. (#687 )

File: site/docs/DocsPage.js (hunk 2 above)
Comment (pradeepnschrodinger) [reply site/docs/DocsPage.js:49]: it should be, but I'll smoke test it.
It's hard to immediately know without prop types defined...

**Comment by pradeepnschrodinger** [comment]
> How does the auto-generate thing works?

@karry08, check out [buildAPIDocs.sh](https://github.com/schrodinger/fixed-data-table-2/blob/2706c1973a86f6a572d4b24ab3343c9864cbf933/build_helpers/buildAPIDocs.sh). It uses [react-docgen](https://github.com/reactjs/react-docgen) internally.

**Pull request merged**